x-pack/packages/kbn-elastic-assistant/impl/assistant/conversations/conversation_settings_management/index.tsx

```diff
@@ -0,0 +1,262 @@
+import { useConversationChanged } from '../conversation_settings/use_conversation_changed';
+import { DEFAULT_PAGE_INDEX, DEFAULT_PAGE_SIZE } from '../../settings/const';
+
+interface Props {
+  actionTypeRegistry: ActionTypeRegistryContract;
+  allSystemPrompts: Prompt[];
+  assistantStreamingEnabled: boolean;
+  connectors: AIConnector[] | undefined;
+  conversationSettings: Record<string, Conversation>;
+  conversationsSettingsBulkActions: ConversationsBulkActions;
+  conversationsLoaded: boolean;
+  defaultConnector?: AIConnector;
+  handleSave: (shouldRefetchConversation?: boolean) => void;
+  isDisabled?: boolean;
+  isFlyoutMode: boolean;
+  onCancelClick: () => void;
+  setAssistantStreamingEnabled: React.Dispatch<React.SetStateAction<boolean>>;
+  setConversationSettings: React.Dispatch<React.SetStateAction<Record<string, Conversation>>>;
+  setConversationsSettingsBulkActions: React.Dispatch<
+    React.SetStateAction<ConversationsBulkActions>
+  >;
+  selectedConversation: Conversation | undefined;
+  onSelectedConversationChange: (conversation?: Conversation) => void;
+}
+
+const ConversationSettingsManagementComponent: React.FC<Props> = ({
+  actionTypeRegistry,
+  allSystemPrompts,
+  assistantStreamingEnabled,
+  connectors,
+  defaultConnector,
+  conversationSettings,
+  conversationsSettingsBulkActions,
+  conversationsLoaded,
+  handleSave,
+  isDisabled,
+  isFlyoutMode,
+  onSelectedConversationChange,
+  onCancelClick,
+  selectedConversation,
+  setAssistantStreamingEnabled,
+  setConversationSettings,
+  setConversationsSettingsBulkActions,
+}) => {
+  const { http } = useAssistantContext();
+  const {
+    isFlyoutOpen: editFlyoutVisible,
+    openFlyout: openEditFlyout,
+    closeFlyout: closeEditFlyout,
+  } = useFlyoutModalVisibility();
+  const [deletedConversation, setDeletedConversation] = useState<ConversationTableItem | null>();
+
+  const {
+    isFlyoutOpen: deleteConfirmModalVisibility,
+    openFlyout: openConfirmModal,
+    closeFlyout: closeConfirmModal,
+  } = useFlyoutModalVisibility();
+
+  const onConversationSelectionChange = useConversationChanged({
+    allSystemPrompts,
+    conversationSettings,
+    conversationsSettingsBulkActions,
+    defaultConnector,
+    setConversationSettings,
+    setConversationsSettingsBulkActions,
+    onSelectedConversationChange,
+  });
+
+  const onEditActionClicked = useCallback(
+    (rowItem: ConversationTableItem) => {
+      openEditFlyout();
+      onConversationSelectionChange(rowItem);
+    },
+    [onConversationSelectionChange, openEditFlyout]
+  );
+
+  const onConversationDeleted = useConversationDeleted({
+    conversationSettings,
+    conversationsSettingsBulkActions,
+    setConversationSettings,
+    setConversationsSettingsBulkActions,
+  });
+
+  const onDeleteActionClicked = useCallback(
+    (rowItem: ConversationTableItem) => {
+      setDeletedConversation(rowItem);
```

Comment:
Doesn't look like there's actually a bug here, but I found it interesting that when you click the delete action it actually removes it from the table before your confirm the action. If you hit cancel it comes back, and if you confirm it then actually sends the REST request, so all is good at the end of the day.

<p align="center">
  <img width="500" src="https://github.com/elastic/kibana/assets/2946766/f8ca8033-b5d1-4c80-906b-674ff0adc272" />
</p>

Reply:
Yeah, this is an existing behaviour from the settings' model. Ideally it shouldn't show the changes before hitting save in the view. Is it alright if we fix this behaviour while removing the settings' model?